src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -82,6 +83,59 @@ inline int power_function(int a, int b)
     return product;
 }
 
+inline void saturate_pixel(Rpp32f pixel, Rpp8u* dst)
+{
+    *dst = RPPPIXELCHECK(pixel);
+}
+
+inline void saturate_pixel(Rpp32f pixel, Rpp8s* dst)
+{
+    *dst = RPPPIXELCHECKI8(pixel);
+}
+
+inline void saturate_pixel(Rpp32f pixel, Rpp32f* dst)
+{
+    *dst = (Rpp32f)pixel;
+}
+
+inline void saturate_pixel(Rpp32f pixel, Rpp16f* dst)
+{
+    *dst = (Rpp16f)pixel;
+}
+
+inline float sinc(float x)
+{
+    x *= PI;
+    if (std::abs(x) < 1e-5f)
+        return 1.0f - x * x * 0.1666666f;  // remove singularity by using Taylor expansion
+    return std::sin(x) / x;
+}
+
+inline void calculate_lanczos_coefficients(float* coeffs, float x)
+{
+    int k = 6;
+    float sum = 0;
+    for(int i=0; i < k; i++)
+    {
+        float xTemp = x - i + 2;
+        coeffs[i] = fabs(xTemp) >= 3 ? 0.0f : (sinc(xTemp)*sinc(xTemp / 3));
+        sum += coeffs[i];
+    }
+    sum = 1.f/sum;
+    for(int i = 0; i < k; i++ )
+        coeffs[i] *= sum;
+}
+
+inline void calculate_cubic_coefficients(Rpp32f* coeffs, Rpp32f x)
+{
+    Rpp32f x2 = x * x;
```

Comment:
Rpp32f xo2 = 0.5f * x;
    Rpp32f xt3o2 = xo2 * 3.0f;
    Rpp32f xp2 = x * x;
    coeffs[0] = (-xo2 + 1) * xp2 - xo2;
    coeffs[1] = (xt3o2 - 2.5) * xp2 + 1;
    coeffs[2] = -(xt3o2 + 2) * xp2 + xo2;
    coeffs[3] = 1.0f - coeffs[0] - coeffs[1] - coeffs[2];

Reply:
Please replace with this and check if unit test is passing